packages/dds/tree/src/test/domains/json/twitter.ts

```diff
@@ -0,0 +1,634 @@
+    profile_image_url_https: string;
+    profile_banner_url?: string;
+    profile_link_color: string;
+    profile_sidebar_border_color: string;
+    profile_sidebar_fill_color: string;
+    profile_text_color: string;
+    profile_use_background_image: boolean;
+    default_profile: boolean;
+    default_profile_image: boolean;
+    following: boolean;
+    follow_request_sent: boolean;
+    notifications: boolean;
+}
+/* eslint-enable */
+
+export interface TwitterJson {
+    statuses: TwitterStatus[];
+    search_metadata: {
+        completed_in: number;
+        max_id: number;
+        max_id_str: string;
+        next_results: string;
+        query: string;
+        refresh_url: string;
+        count: number;
+        since_id: number;
+        since_id_str: string;
+    };
+}
+
+/**
+* Generates a TwitterJson object as closely as possible to a specified byte size.
+* The generated json will as close to the specified size but will almost always be slightly less.
+* @param sizeInBytes - size to generate json object
+* @param includeUnicode - true to include unicode in any strings within the json
+* @param allowOversize - Allows the json to go over the sizeInBytes limit. If enabled, the
+* generated json may be closer to the desired byte size but there is a risk of exceeding the inputted byte limit
+* @returns TwitterJson
+*/
+export function generateTwitterJsonByByteSize(sizeInBytes: number, allowOversize: boolean, seed = 1) {
+    const random = makeRandom(seed);
+    const textFieldMarkovChain = new SpaceEfficientMarkovChain(random, getTwitterJsonTextFieldMarkovChain());
+    const userDescFieldMarkovChain = new SpaceEfficientMarkovChain(random, getTwitterJsonUserDescFieldMarkovChain());
+    const basicJapaneseAlphabetString = getBasicJapaneseAlphabetString();
+    const twitterJson: TwitterJson = {
+        statuses: [],
+        search_metadata: {
+            completed_in: 0.087,
+            max_id: 505874924095815700,
+            max_id_str: "505874924095815681",
+            next_results: "?max_id=505874847260352512&q=%E4%B8%80&count=100&include_entities=1",
+            query: "%E4%B8%80",
+            refresh_url: "?since_id=505874924095815681&q=%E4%B8%80&include_entities=1",
+            count: 100,
+            since_id: 0,
+            since_id_str: "0",
+        },
+    };
+
+    let currentJsonSizeInBytes = getSizeInBytes(twitterJson);
+    while (currentJsonSizeInBytes < sizeInBytes) {
+        const twitterStatus = generateTwitterStatus(
+            "standard", random, textFieldMarkovChain, userDescFieldMarkovChain, basicJapaneseAlphabetString,
+        );
+        const nextStatusSizeInBytes = getSizeInBytes(twitterStatus);
+        if (!allowOversize && currentJsonSizeInBytes + nextStatusSizeInBytes > sizeInBytes) {
+            break;
+        }
+        twitterJson.statuses.push(twitterStatus);
+        currentJsonSizeInBytes += nextStatusSizeInBytes;
+    }
+
+    return twitterJson;
+}
```

Comment:
FWIW, I only have plans for specifying the num statuses (or num events/performances for CITM).